builder/src/main/java/cz/xtf/builder/db/MsSQL.java

```diff
@@ -0,0 +1,114 @@
+package cz.xtf.builder.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import cz.xtf.builder.builders.pod.ContainerBuilder;
+import cz.xtf.builder.builders.pod.PersistentVolumeClaim;
+import cz.xtf.core.config.OpenShiftConfig;
+import cz.xtf.core.image.Image;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import io.fabric8.kubernetes.api.builder.Visitor;
+import io.fabric8.kubernetes.api.model.ServiceAccountBuilder;
+import io.fabric8.openshift.api.model.SecurityContextConstraintsBuilder;
+
+/**
+ * Note that MsSQL Server database needs to run with custom user id on Openshift (0 should do the trick)*
+ * Also, the MsSQL image doesn't provide any option to add custom database and user on setup.
+ *
+ * By using this class you are accepting the Microsoft's EULA: https://go.microsoft.com/fwlink/?linkid=857698
+ */
+public class MsSQL extends AbstractSQLDatabase {
+
+    private final String ANYUID_SERVICE_ACCOUNT = "anyuid";
+    private final long USER_ID = 0;
+
+    public MsSQL() {
+        super("sa", "myMsSQLPassword123", "master", "MSSQL", "/var/lib/mssql/data");
+    }
+
+    public MsSQL(boolean withLivenessProbe, boolean withReadinessProbe) {
+        super("sa", "myMsSQLPassword123", "master", "MSSQL", "/var/lib/mssql/data", withLivenessProbe, withReadinessProbe);
+    }
+
+    public MsSQL(PersistentVolumeClaim pvc) {
+        super("sa", "myMsSQLPassword123", "master", "MSSQL", "/var/lib/mssql/data", pvc);
+    }
+
+    public MsSQL(PersistentVolumeClaim pvc, boolean withLivenessProbe, boolean withReadinessProbe) {
+        super("sa", "myMsSQLPassword123", "master", "MSSQL", "/var/lib/mssql/data", pvc, withLivenessProbe, withReadinessProbe);
+    }
+
+    public MsSQL(String username, String password, String dbName) {
+        super(username, password, dbName, "MSSQL", "/var/lib/mssql/data");
+    }
+
+    public MsSQL(String symbolicName, boolean withLivenessProbe, boolean withReadinessProbe) {
+        super("sa", "myMsSQLPassword123", "master", symbolicName, "/var/lib/mssql/data", withLivenessProbe, withReadinessProbe);
+    }
+
+    @Override
+    protected void configureContainer(ContainerBuilder containerBuilder) {
+        super.configureContainer(containerBuilder);
+        
+        //Setting up the anyuid service account
+        try (OpenShift master = OpenShifts.master()) {
+            if (master.getServiceAccount(ANYUID_SERVICE_ACCOUNT) == null) {
+                master.createServiceAccount(
+                        new ServiceAccountBuilder().withNewMetadata().withName(ANYUID_SERVICE_ACCOUNT).endMetadata().build());
+            }
+        }
+        try (OpenShift admin = OpenShifts.admin()) {
```

Comment:
This is common to both MsSQL and OracleXE. Have you tried to  avoid using `admin` ? It can happen  we will be forced to run this code on cluster without `admin`. Would that be possible?  If not can you document which permission regular `master` user will be missing?

Reply:
Regular master is missing permissions to add security context constraint. I added this info in the comment before the classes.